src/error/MatterError.ts

```diff
@@ -0,0 +1,146 @@
+
+export class BusyStatusResponseError extends StatusResponseError {}
+StatusResponseError.registerErrorClass(StatusCode.Busy, BusyStatusResponseError);
+
+export class UnsupportedClusterStatusResponseError extends StatusResponseError {}
+StatusResponseError.registerErrorClass(StatusCode.UnsupportedCluster, UnsupportedClusterStatusResponseError);
+
+export class NoUpstreamSubscriptionStatusResponseError extends StatusResponseError {}
+StatusResponseError.registerErrorClass(StatusCode.NoUpstreamSubscription, NoUpstreamSubscriptionStatusResponseError);
+
+export class NeedsTimedInteractionStatusResponseError extends StatusResponseError {}
+StatusResponseError.registerErrorClass(StatusCode.NeedsTimedInteraction, NeedsTimedInteractionStatusResponseError);
+
+export class UnsupportedEventStatusResponseError extends StatusResponseError {}
+StatusResponseError.registerErrorClass(StatusCode.UnsupportedEvent, UnsupportedEventStatusResponseError);
+
+export class PathsExhaustedStatusResponseError extends StatusResponseError {}
+StatusResponseError.registerErrorClass(StatusCode.PathsExhausted, PathsExhaustedStatusResponseError);
+
+export class TimedRequestMismatchStatusResponseError extends StatusResponseError {}
+StatusResponseError.registerErrorClass(StatusCode.TimedRequestMismatch, TimedRequestMismatchStatusResponseError);
+
+export class FailsafeRequiredStatusResponseError extends StatusResponseError {}
+StatusResponseError.registerErrorClass(StatusCode.FailsafeRequired, FailsafeRequiredStatusResponseError);
+
+
+export function isMatterError(e: unknown): e is MatterError {
+    return e instanceof MatterError;
+}
+
+type MatterErrorHandler<T> = (error: MatterError) => T | undefined;
+
+export function tryCatch<T>(codeBlock: () => T, errorType: typeof MatterError, fallbackValue: MatterErrorHandler<T> | T): T {
+    try {
+        return codeBlock();
+    }
+    catch (error) {
+        if (isMatterError(error) && error instanceof errorType) {
+            if (typeof fallbackValue === "function") {
+                const computedFallbackValue = (fallbackValue as MatterErrorHandler<T>)(error);
+                if (computedFallbackValue !== undefined) return computedFallbackValue;
+            } else {
+                return fallbackValue;
+            }
+        }
+        throw error;
+    }
+}
+
+export async function tryCatchAsync<T>(codeBlock: () => Promise<T>, errorType: typeof MatterError, fallbackValue: MatterErrorHandler<T> | T): Promise<T> {
+    try {
+        return await codeBlock();
+    }
+    catch (error) {
+        if (isMatterError(error) && error instanceof errorType) {
+            if (typeof fallbackValue === "function") {
+                const computedFallbackValue = (fallbackValue as MatterErrorHandler<T>)(error);
+                if (computedFallbackValue !== undefined) return Promise.resolve(computedFallbackValue);
+            } else {
+                return Promise.resolve(fallbackValue);
```

Comment:
I don't sync you need a Promise.resolve here since the function is declared as async